snuba/web/views.py

```diff
@@ -280,11 +284,37 @@ def unqualified_query_view(*, timer: Timer) -> Union[Response, str, WerkzeugResp
 def rpc(*, name: str, version: str) -> Response:
     try:
         endpoint = RPCEndpoint.get_from_name(name, version)()  # type: ignore
+    except AttributeError:
+        return convert_rpc_exception_to_proto(
+            RPCRequestException(
+                status_code=404,
+                message=f"endpoint {name} with version {version} does not exist (did you use the correct version and capitalization?)",
+            )
+        )
+
+    try:
         deserialized_protobuf = endpoint.parse_from_string(http_request.data)
+    except DecodeError as e:
+        return convert_rpc_exception_to_proto(
+            RPCRequestException(
+                status_code=400,
+                message=f"protobuf gave a decode error {e} (are all fields set and the correct types?)",
+            )
+        )
+
+    try:
         res = endpoint.execute(deserialized_protobuf)
         return Response(res.SerializeToString())
-    except BadSnubaRPCRequestException as e:
-        return Response(str(e), status=400)
+    except (RPCRequestException, QueryException) as e:
+        exc_proto = convert_rpc_exception_to_proto(e)
+        return Response(exc_proto.SerializeToString(), status=exc_proto.code)
+    except Exception as e:
+        return convert_rpc_exception_to_proto(
+            RPCRequestException(
+                status_code=500,
+                message=f"internal error occurred while executing this RPC call: {e}",
+            )
+        )
```

Comment:
I would like to have as little logic in this layer as possible and as much as possible moved into the `RPCEndpoint` class. This is doing too much in the flask layer